src/topology/topological_fiber_bundle.lean

```diff
@@ -979,3 +988,214 @@ begin
+prod.ext (e.coe_fst' ex).symm rfl
+
+protected lemma prebundle_trivialization.eq_on (e : prebundle_trivialization F proj) :
+  eq_on (prod.fst ∘ e) proj e.source :=
+λ x hx, e.coe_fst hx
+
+lemma prebundle_trivialization.proj_symm_apply (e : prebundle_trivialization F proj) {x : B × F}
+  (hx : x ∈ e.target) : proj (e.to_local_equiv.symm x) = x.1 :=
+begin
+  have := (e.coe_fst (e.to_local_equiv.map_target hx)).symm,
+  rwa [← e.coe_coe, e.to_local_equiv.right_inv hx] at this
+end
+
+lemma prebundle_trivialization.proj_symm_apply' (e : prebundle_trivialization F proj) {b : B}
+  {x : F} (hx : b ∈ e.base_set) : proj (e.to_local_equiv.symm (b, x)) = b :=
+e.proj_symm_apply (e.mem_target.2 hx)
+
+lemma prebundle_trivialization.apply_symm_apply (e : prebundle_trivialization F proj)
+  {x : B × F} (hx : x ∈ e.target) : e (e.to_local_equiv.symm x) = x :=
+e.to_local_equiv.right_inv hx
+
+lemma prebundle_trivialization.apply_symm_apply' (e : prebundle_trivialization F proj)
+  {b : B} {x : F} (hx : b ∈ e.base_set) : e (e.to_local_equiv.symm (b, x)) = (b, x) :=
+e.apply_symm_apply (e.mem_target.2 hx)
+
+@[simp, mfld_simps] lemma prebundle_trivialization.symm_apply_mk_proj
+  (e : prebundle_trivialization F proj) {x : Z} (ex : x ∈ e.source) :
+  e.to_local_equiv.symm (proj x, (e x).2) = x :=
+by rw [← e.coe_fst ex, prod.mk.eta, ← e.coe_coe, e.to_local_equiv.left_inv ex]
+
+@[simp, mfld_simps] lemma prebundle_trivialization.preimage_symm_proj_base_set
+  (e : prebundle_trivialization F proj) :
+  (e.to_local_equiv.symm ⁻¹' (proj ⁻¹' e.base_set)) ∩ e.target  = e.target :=
+begin
+  refine inter_eq_right_iff_subset.mpr (λ x hx, _),
+  simp only [mem_preimage, local_equiv.inv_fun_as_coe, e.proj_symm_apply hx],
+  exact e.mem_target.mp hx,
+end
+
+@[simp, mfld_simps] lemma prebundle_trivialization.preimage_symm_proj_inter
+  (e : prebundle_trivialization F proj) (s : set B) :
+  (e.to_local_equiv.symm ⁻¹' (proj ⁻¹' s)) ∩ e.base_set.prod univ = (s ∩ e.base_set).prod univ :=
+begin
+  refine subset.antisymm_iff.mpr ⟨(λ x hx, _), (λ x hx, mem_inter _ _)⟩,
+  { rw [←e.target_eq] at hx,
+    simp only [mem_inter_iff, mem_preimage, e.proj_symm_apply hx.2] at hx,
+    simp only [mem_inter_eq, and_true, mem_univ, mem_prod],
+    exact ⟨hx.1, e.mem_target.mp hx.2⟩, },
+  { simp only [mem_inter_eq, and_true, mem_univ, mem_prod, e.mem_target.symm] at hx,
+    simp only [mem_preimage, e.proj_symm_apply hx.2],
+    exact hx.1, },
+  { rw [←inter_univ univ, ←prod_inter_prod, mem_inter_eq] at hx,
+    exact hx.2, }
+end
+
+variable (F)
+
+/-- This structure permits to define a fiber bundle when trivializations are given as local
+equivalences but there is not yet a topology on the total space. The total space is hence given a
+topology in such a way that there is a fiber bundle structure for which the local equivalences
+are also local homeomorphism and hence local trivializations. -/
+@[nolint has_inhabited_instance]
```

Comment:
Actually this `nolint` is not necessary.

Reply:
Then you can remove it. Whenever you see something that can be fixed, you can just fix it right away, it can only help the reviewers :-)

Reply:
I addeded it because this would be the trivial prebundle and since no one would want to defind the (already defined) trivial bundle through this construction I don't see why we should have this in mathlib apart to avoid the nolint line.

Reply:
This `has_inhabited_instance` is useful to tell people "give at least one nontrivial example of your objects". You certainly don't *have* to (you can `nolint` as you did) but it can help spot mistakes.

Reply:
What would _nontrivial_ mean here?

Reply:
Any one you like! The trivial bundle definitely counts as nontrivial here :-)

Reply:
Actually I was wrong. Removing the nolint would mean that the type is inhabited no matter what the total space is and this is wrong (e.g. when the fibers have different cardinalities).